pkgs/applications/networking/cluster/k3s/docs/USAGE.md

```diff
@@ -64,3 +64,37 @@ Tip: If you run into connectivity issues between nodes for specific applications
 K3s has a config setting `prefer-bundled-bin` (and CLI flag `--prefer-bundled-bin`) that makes k3s use binaries from the `/var/lib/rancher/k3s/data/current/bin/aux/` directory, as unpacked by the k3s binary, before the system `$PATH`.
 This works with the official distribution of k3s but not with the package from nixpkgs, as it does not bundle the upstream binaries from [`k3s-root`](https://github.com/k3s-io/k3s-root) into the k3s binary.
 Thus the `prefer-bundled-bin` setting **cannot** be used to work around issues (like [this `mount` regression](https://github.com/util-linux/util-linux/issues/3474)) with binaries used/called by the kubelet.
+
+### Building from a different source
+
+Because the package is split into multiple derivations and the build process is generally more complex, it is not very obvious how to build k3s from a different source (fork or arbitrary commit).
+
+To build k3s from a different source, you must use `.override` together with `overrideBundleAttrs` (for the k3sBundle derivation) and another `.overrideAttrs` (for the final derivation):
+
+```nix
+{ fetchgit, k3s }:
+let
+  k3sRepo = fetchgit {
+    url = "https://github.com/k3s-io/k3s";
+    rev = "99d91538b1327da933356c318dc8040335fbb66c";
+    hash = "sha256-vVqZzVp0Tea27s8HDVq4SgqlbHBdZcFzNKmPFi0Yktk=";
+  };
+  vendorHash = "sha256-jrPVY+FVZV9wlbik/I35W8ChcLrHlYbLAwUYU16mJLM=";
+in
+(k3s.override {
+  overrideBundleAttrs = {
+    src = k3sRepo;
+    inherit vendorHash;
+  };
+}).overrideAttrs
+  (_: {
+    src = k3sRepo;
+    inherit vendorHash;
+  })
```

Comment:
Nit

```suggestion
  {
    src = k3sRepo;
    inherit vendorHash;
  }
```

Reply:
Lol didn't know this was valid :eyes: thx :laughing: